### ISSUE TITLE
Launch Pads and Possible General Stuff

### ISSUE THREAD

**Issue by GitBobCat** [Anuken/Mindustry#5289]
Platform: Linux Mint

Build: 126.2

Issue: This has cropped up a couple of times in bug reports. The problem is kind of you can't stop them doing their thing so all of your resources disappear into the ether. In one response you described it as not being a bug so perhaps it is a feature but it happens. Elsewhere you mention that the game takes an average over some time, several minutes, and then runs with it in the future. In addition to the launch pads being silly it is noticeable that sectors are prone to hanging which I'll guess is down to the same averaging. This is me watching sectors with launch pads trying to figure out what is going on and possibly not sitting on the sector for long enough. There have also been a number of occasions where I've left a sector under attack to check on launch pads only to discover that something like a sneak attack has occurred. Most unfair.

Waving my wet finger in the air I'm going to guess that whilst the game supposedly operates sectors in background what it is actually doing is taking an average for a sector whilst you are on it and running with that in the future. As a result it is prone to getting a horribly wrong answer. You were launching 3000 thorium per minute for four minutes, you turn off your launch pads and immediately leave the sector. The program based on the average assumes you are still launching 3000 thorium per minute and all of your thorium disappears.

Apologies for the ramble.

However might a possible solution be to delay the exit for say 5 seconds and run the sector simulation at maximum speed to gain a better estimate of the average rates of what is going on. I'm guessing, probably incorrectly, that things run a bit faster when you are no longer updating graphics and stuff.

Failing that or even better, perhaps according to some sort of priority, would it be possible to have the program spawn, thread, sectors in background to perform the same sort of action such that the averages are more up to date or consistent. I realise that might be a big ask but whilst I don't know much about programming I have heard of such things. Perhaps as a game option. 

Steps to reproduce: Just playing the game.

Link(s) to mod(s) used: No Mods.

Save file: In this case I don't think you need one. It's known to happen. As as aside however under Linux Mint files are saved under .wine/dosdevices/z:/proc/number and get deleted when you exit the game. I assume /proc/number is a process that gets totally killed on exit.

If you remove the line above without reading it properly and understanding what it means, I will reap your soul. Even if you're playing on someone's server, you can still save the game to a slot.

(Crash) logs: No crashes.

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [X] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [X] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by Anuken** [comment]
Again, this isn't a bug.

> run the sector simulation at maximum speed

It doesn't matter how much the simulation runs, it'll still have the same average values.

> would it be possible to have the program spawn, thread-

Absolutely not.

**Comment by GitBobCat** [comment]
Oh. Uhm kind of OK. It's not as if. Mumble. Apologies for the following.

If the average is the same then that means the program implements the full theory thing about process control to work it out. Just to be boring I did a computer simulation of chromatography based on plate theory. It's the closest I've got to what you are doing. Perhaps I might have come up with a better solution if I knew about process control or maybe not. I had to deal with some weird isotherms.

I guess, for whatever reason, it remains as a serious pain in the elbow when you set up Salt Flats as your main launch point then fill it up from the elsewhere's and launch that to another place then find out that the elsewhere's no longer have any resources and Salt Flats is fully depleted as well.

Am I doing it wrong?
